wasm/ro/chapter_2.yaml

```
@@ -0,0 +1,244 @@
+
+    console.log(text) // Hi!
+
+
+    const utf8enc = new TextEncoder("utf-8");
+
+    let text_bytes = utf8enc.encode(text);
+
+    console.log(text_bytes) 
+
+    // Uint8Array(3) [72, 105, 33]
+
+    ```
+- title: Înregistrarea textului
+  code: "https://webassembly.studio/?embed&f=1gxcr004p3x"
+  content_markdown: >
+    Haideți să explorăm un exemplu simplu de înregistrare a unui text dintr-un
+    program WebAssembly.
+
+
+    Este necesar:
+
+    1. Să creăm un text compatibil cu UTF-8 în memoria programului nostru.
+
+    2. Să determinăm lungimea octeților textului nostru.
+
+    3. Să găsim o modalitate de a transmite indicele de început al primului octet
+    și lungimea în octeți a textului către browser-ul gazdei pentru a putea
+    apela `console.log`.
+
+
+    Iată un exemplu de cum ar arăta acea funcție JavaScript de primire:
+
+    ```JavaScript
+
+    wasm_log(start,len) {
+      // extrage textul din memorie și lungimea
+      const utf8dec = new TextDecoder("utf-8");
+      let buffer = module.instance.exports.memory.buffer;
+      let memory = new Uint8Array(buffer);
+      let text = utf8dec.decode(memory.subarray(start,start+len));
+      console.log(text);
+    }
+
+    ```
+- title: Citirea textului
+  code: "https://webassembly.studio/?embed&f=ubmxmavgf2"
+  content_markdown: >
+    Să explorăm conceptul opus. Imaginați-vă că dorim să dăm niște text
+    unui program WebAssembly.
+
+
+    Este necesar:
+
+    1. Să determinăm lungimea în bytes a textului pe care vrem să-l dăm.
+
+    2. Să alocăm suficientă memorie pentru a stoca textul nostru.
+
+    3. Să copiem bytes în memoria programului nostru la începutul spațiului
+    alocat.
+
+    4. Să anunțăm programul nostru că am pus textul în memoria sa la un
+    anumit indice de început și lungime în bytes.
+
+
+    Iată un exemplu de cum ar arăta acea inițializare:
+
+    ```JavaScript
+    
+    // Transformă "Ferris" în bytes
+
+    const utf8enc = new TextEncoder("utf-8");
+
+    let text = "Ferris";
+
+    let text_bytes = utf8enc.encode(text);
+
+
+    // Alocă suficient spațiu pentru text
+
+    let len = text_bytes.length;
+
+    let start = module.instance.exports.wasm_malloc(len);
+
+
+    // Pune textul în memoria programului WebAssembly
+
+    let buffer = module.instance.exports.memory.buffer;
+
+    let memory = new Uint8Array(buffer);
+
+    memory.set(text_bytes, start);
+
+
+    // Rulează programul
+
+    module.instance.exports.main(start,len);
+
+    ```
+- title: Descoperirea reprezentărilor
+  content_markdown: >
+    Vi s-ar putea părea enervant să trebuiască să dați lungimea alături de
+    indicele de început al textului.
+
+
+    Considerați această situație drept o oportunitate de a reflecta asupra
+    convențiilor utile, precum *string-urile din C*,
+
+    care declară terminarea unui text ca o valoare nulă prin caracterul `\0`.
+
+
+    Nu doriți să pasați peste tot o structură de date "învelită" ca octeți?
```

```suggestion
    Nu doriți să transmiteți peste tot o structura plină de octeți?
```

---

Mulțumesc pentru semnalare! Contextul se referea la reprezentarea structurii, nu la conținutul ei. Așadar, cred că o modificare adecvată ar fi:
```suggestion
    Nu doriți să transmiteți peste tot o structură de date reprezentată sub formă de octeți?
```